### PR TITLE
Added DatabaseReplica resource support

### LINKED ISSUE #224: Read-only Database Replica Resource
With the addition of support for DigitalOcean Managed Databases, we should extend support for related resources such as read-only replicas.

Proposed format:

```hcl
resource "digitalocean_database_replica" "read-01" {
  cluster_id = "${digitalocean_database.foo.id}"
  name       = "read-replica-01"
  region     = "nyc3"
  size       = "db-s-2vcpu-4gb"
}
```

API: https://developers.digitalocean.com/documentation/v2/#create-a-read-only-replica

Related: https://github.com/terraform-providers/terraform-provider-digitalocean/issues/189

### PR DESCRIPTION
**Pull request by Zyqsempai**
Fixes #224 
Added DatabaseReplica resource support + tests

### CODE REVIEW

**Comment by Zyqsempai** [comment]
@andrewsomething PTAL

**Comment by Zyqsempai** [comment]
@andrewsomething Done!

File: digitalocean/resource_digitalocean_database_replica.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+	}
+
+	replica, err = waitForDatabaseReplica(client, clusterId, "online", replica.Name)
+	if err != nil {
+		return fmt.Errorf("Error creating DatabaseReplica: %s", err)
+	}
+
+	log.Printf("[INFO] DatabaseReplica Name: %s", replica.Name)
+
+	return resourceDigitalOceanDatabaseClusterRead(d, meta)
```
Comment (andrewsomething) [inline digitalocean/resource_digitalocean_database_replica.go]: Missed this one on my first pass.

```suggestion
	return resourceDigitalOceanDatabaseReplicaRead(d, meta)
```

**Comment by Zyqsempai** [comment]
@andreaso Done!

**Comment by andrewsomething** [comment]
One thing to note: I did make some additional changes around importing. Since you need both the cluster's ID and the replicas name to access an existing replica, we can't just use the resource passthrough for imports. You need to pass both values in, eg:

```
terraform import digitalocean_database_replica.read-replica 245bcfd0-7f31-4ce6-a2bc-475a116cca97,read-replica
```

So it required a custom import function:

https://github.com/terraform-providers/terraform-provider-digitalocean/blob/master/digitalocean/resource_digitalocean_database_replica.go#L153-L162

I also added an acceptance test for that: https://github.com/terraform-providers/terraform-provider-digitalocean/blob/master/digitalocean/import_digitalocean_database_replica_test.go

**Pull request merged**